fixed_bed_model.py

```diff
@@ -0,0 +1,2630 @@
+        m.b_a = Param(initialize=b_a0*exp(Q_sta/(RPV*T0)*(T0/m.temp-1))) 
+
+        # LHS: dimensionless
+        # RHS: dimensionless * exp( [kJ/mol]/[kJ/mol/K * K] ) = [dimensionless] * exp([dimensionless ])
+        m.n_a = Param(initialize=n_a1*exp(E_na/(RPV*T0)*(T0/m.temp-1)))
+
+        #m.inv_n_a = Param(initialize=1/m.n_a)
+        inv_n_a = 1/m.n_a
+
+        # LHS: dimensionless
+        # RHS: dimensionless + K/K
+        m.K_eq = Param(initialize=exp(Ka + Kb/m.temp))
+
+        ### Physical adsorption isotherm
+
+        # LHS: bar-1
+        # RHS: bar-1 * (kJ/mol)/(kJ/mol/K * K)
+        m.b_b = Param(initialize = b_b0*exp(Q_stb/(RPV*T0)*(T0/m.temp-1)))
+
+        # LHS: mol/kg
+        # RHS: mol/kg * 1
+        m.nmax_p = Param(initialize=nmax_p1*(exp(Kc+Kd/m.temp)/(1+exp(Kc+Kd/m.temp))))
+
+        # LHS: mol/kg
+        # RHS: mol/kg * 1 / 1 = mol/kg
+        m.nmax_c = Param(initialize=n_max*m.K_eq/(1+m.K_eq))
+
+        # Linear pressure adsorption when P < Plinear 
+        # According to [Hughes et al., 2021]
+        # LHS: mol/kg 
+        # RHS: mol/kg * ((1/bar * bar )/ (bar^-1 * 10bar))
+        nplin_num = (m.b_a*plin)**(inv_n_a)
+        nplin_de = 1+(m.b_a*plin)**(inv_n_a)
+        m.nplin = Param(initialize=m.nmax_c*(nplin_num/nplin_de))
+                 
+    return m
+
+
+    
+def bounds_velocity(m, z, t):
+    ''' Add bounds to velocity [cm/s]
+    '''
+    if t == 0.0:
+        #return (1E-4, 0.3)  # bounds for 313K, 0.15 
+        return (1E-2, 30)  # bounds for DoE problem
+    else:
+        #return (0.8,1.4)  # bounds for 313K, 0.15 
+        return (0.01, 5.0)  # bounds for DoE problem
+    
+def bounds_velocity_pert(m, j, z, t):
+    ''' Add bounds to velocity [cm/s]
+    '''
+    # Change to cm/s
+    if t == 0.0:
+        #return (1E-4, 0.3)  # bounds for 313K, 0.15 
+        return (1E-2, 30) # bounds for DoE problem
+    else:
+        #return (0.8,1.4)  # bounds for 313K, 0.15 
+        return (0.01, 5.0) # bounds for DoE problem
+
+def breakthrough_bounds(z,t):
+    ''' Check if position and time coordinates in region with
+    no breakthrough.
+    
+        We draw a line between (t = 50% breakthrough, end of bed) and
+        (t=0, z = 25% bed length). Anything "above" this line we designate as in the 
+        region of no breakthrough and return True. We impose tighter bounds if true.
+    '''
+
+    t50 = 1500 # 50% breakthrough time [s]
+    t0_intercept = 0.4
+
+    # return (t0_intercept - 1)/t50*value(t) + t0_intercept > value(z)/Ngrid
+    return False
+
+    
+def den_bounds(m,c,t,z):
+    '''[mol/m3]
+    '''
+    if breakthrough_bounds(z,t):
+        if c == 'N2':
+            return (0.9*tden_low, tden_high)
+        elif c == 'CO2':
+            return (small_bound, 0.1*tden_high)
+        else:
+            return (small_bound, tden_high)
+    else:
+        return (small_bound, tden_high)
+    
+def den_bounds_pert(m,j,c,t,z):
+    if breakthrough_bounds(z,t):
```

Comment:
Add comment. What does `_pert` mean in the function name? How is this different than the previous function name?